Add list of federated extensions for the demo site (#84)
jtpio/jupyterlite: docs/deploying.md

- _TBD: installing `@jupyterlite/builder` from `npmjs.com`_
- downloading nightly/work-in-progress builds from [GitHub actions]
- cloning/forking the repository and doing a [development build](../contributing.md)

[github actions]: https://github.com/jtpio/jupyterlite/actions
[releases]: https://github.com/jtpio/jupyterlite/releases
[pypi]: https://pypi.org/project/jupyterlite/

```{hint}
It is recommended to put these files under revision control. See [Configuring](./configuring.md)
for what you can configure in your JupyterLite.
```

## Build Tools

### WebPack

> TBD

### sphinx

[Sphinx] is the workhorse of documentation of not only the scientific Python
documentation community, but also the broader Python ecosystem, and many languages
beyond it. It is well adapted to building sites of any size, and tools like [myst-nb]
enable make it very palletable to include executable, and even interactive, content.

JupyterLite assets can be copied to the default static directory in `conf.py`, e.g.
`docs/_static` with [`html_static_path`](#html_static_path), or replace the entire site
with [`html_extra_path`](#html_extra_path)

### `html_static_path`

> This searach path can be merge several layers deep, such that your theme assets, the

Author: ```suggestion
This search path can be merged several layers deep, such that your theme assets, the
```

"gold master" JupyterLite assets, and any customizations you wish to make are combined.

```python
html_static_path = [
    "_static",
    "../upstream-jupyterlite",
    "../my-jupyterlite"        # <- these "win"
]
```

The composite directory will end up in `docs/_build/_static`.

```{hint}

> See the JupyterLite [conf.py] for an example approach, though it likely a good

Author: ```suggestion
See the JupyterLite [conf.py] for an example approach, though it's likely a good
```

python -m http.server -b 127.0.0.1
```

> If you are using a recently-released Python 3.7+, this will correctly server

Author: ```suggestion
If you are using a recently-released Python 3.7+, this will correctly serve
```

### Binder

A JupyterLite can be deployed behind `jupyter-server-proxy` using any
[local server](#local) method. This is a good way to preview deployment interactively of
a e.g. Lab extension that can work in both the "full" binder experience, and as a static
preview.

```{hint}
See the JupyterLite [binder configuration] for an example.
```

[binder configuration]: https://github.com/jtpio/jupyterlite/tree/main/.binder

### ReadTheDocs

The [Sphinx](#sphinx) deployment approach will work almost transparently with
[ReadTheDocs](https://readthedocs.org), for the small price of a `.readthedocs.yml` file
in the root of your repository.

```{hint}

> See the JuptyerLite [.readthedocs.yml] for an example.

Author: ```suggestion
See the JupyterLite [.readthedocs.yml] for an example.
```
